core/src/main/scala/org/apache/spark/util/ManualClock.scala

```diff
@@ -27,6 +27,7 @@ package org.apache.spark.util
 private[spark] class ManualClock(private var time: Long) extends Clock {
 
   private var _isWaiting = false
+  private var _readyForFirstPeek = false
```

Comment:
Its really not obvious on what purpose does _readyForFirstPeek server over _isWaiting.